src/dpl/CMakeLists.txt

```diff
@@ -81,3 +81,29 @@ add_custom_target(opendp_tags etags -o TAGS
   ${OPENDP_HOME}/include/dpl/*.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
 )
+
+if (Python3_FOUND AND BUILD_PYTHON)
+  swig_lib(NAME          dpl_py
+           NAMESPACE     dpl
+           LANGUAGE      python
+           I_FILE        src/Opendp-py.i
+           SWIG_INCLUDES ${PROJECT_SOURCE_DIR}/include/dpl
+                         ${ODB_HOME}/src/swig/common
+                         ${ODB_HOME}/src/swig/python
+           SCRIPTS       ${CMAKE_CURRENT_BINARY_DIR}/dpl_py.py
+  )
+
+  target_include_directories(dpl_py
+    PUBLIC
+      include
+  )
+
+  target_link_libraries(dpl_py
+    PUBLIC
+      gui
+      odb
+      OpenSTA
+      utl
+  )
```

Comment:
Can you just say
```
target_link_libraries(dpl_py
    PUBLIC
      dpl
  )
```

Reply:
So, as written, this doesn't work. You need to add odb, ie
```
target_link_libraries(dpl_py
    PUBLIC
      dpl
      odb
  )
```
The include file Opendp.h is used as the %include file in the -py.i file and this file heavily depends on odb. Generally, my strategy here has been to just mirror what is done for the tcl wrapping. Is there a better way than just trial and error (and on my part that is mostly error) to find out what is superfluous? Anyway, this does work and simplifies the CMakeLists.txt file. Will upload that shortly

Reply:
That's annoying.  Swig gets its include path by a different mechanism than c++ so I guess this is hard to avoid.  I suspect odb is the main case where we include a .i from another module.